File: tests/code/corpus/langs/python/shapes.py
"""Geometry helpers used by the parsing corpus."""
import math
from dataclasses import dataclass


class Shape:
    def area(self):
        return 0.0

    def describe(self):
        return f"shape of area {self.area()}"


class Circle(Shape):
    def __init__(self, radius):
        self.radius = radius

    def area(self):
        return math.pi * self.radius ** 2


@dataclass
class Point:
    x: float
    y: float


def make_circle(radius):
    return Circle(radius)


def total_area(shapes):
    return sum(shape.area() for shape in shapes)

File: tests/code/corpus/langs/ruby/shapes.rb
require "json"

module Geometry
  class Shape
    def area
      0.0
    end

    def describe
      "shape of area #{area}"
    end
  end

  class Circle < Shape
    def initialize(radius)
      @radius = radius
    end

    def area
      3.14159 * @radius * @radius
    end
  end
end

def make_circle(radius)
  Geometry::Circle.new(radius)
end

def total_area(shapes)
  shapes.sum { |shape| shape.area }
end
